programs/mpl-core/src/state/update_authority.rs

```diff
@@ -77,12 +76,7 @@ impl UpdateAuthority {
 
                 if let Ok((authority, _, _)) = maybe_update_delegate {
                     if assert_collection_authority(&collection, authority_info, &authority).is_err()
-                        && assert_collection_authority(
-                            &collection,
-                            authority_info,
-                            &Authority::UpdateAuthority,
-                        )
-                        .is_err()
+                        && authority_info.key != &collection.update_authority
```

Comment:
what was the intention of this change?

Reply:
As I studied this whole code block I believe the intention was just to allow for `collection.update_authority` to still approve when the update delegate is present but not being used.

I think it is more clearly done by just checking directly like this, rather than calling `assert_collection_authority` a second time but with a hardcoded `authority` parameter.

This then also matches the check a few lines down on 84.

The reason I changed it here is because I use similar logic in `update`.